### PR TITLE
Use Connext 6 for Rolling and beyond on RHEL.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
This is a continuation of #614 which adds support for RHEL while retaining the ability to build using 5.3.1 for Foxy and Galactic.

### CODE REVIEW

File: linux_docker_resources/Dockerfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ ARG BRIDGE=false
 ARG INSTALL_CONNEXT_DEBS=false
 ARG PLATFORM=x86
 ARG ROS1_DISTRO=noetic
+ARG ROS_DISTRO=rolling
```
Comment (nuclearsandwich) [inline linux_docker_resources/Dockerfile:6]: Although this build argument is not yet used, and is not strictly necessary for this patch, I felt it worthwhile to apply the argument to both Ubuntu and RHEL docker builds. If preferred I could walk this back though.

File: linux_docker_resources/Dockerfile-RHEL [hunk 2]
```diff
@@ -119,8 +120,21 @@ RUN dnf install \
     --refresh -y
 
 # Get and install the RTI web binaries.
-RUN cd /tmp && curl --silent https://s3.amazonaws.com/RTI/Bundles/5.3.1/Evaluation/rti_connext_dds_secure-5.3.1-eval-x64Linux3gcc5.4.0.tar.gz | tar -xz
-RUN cd /tmp && tar -xvf /tmp/openssl-1.0.2n-target-x64Linux3gcc5.4.0.tar.gz
+# Connext 5.3.1 for galactic and earlier.
+RUN if [ "$ROS_DISTRO" = "foxy" -o "$ROS_DISTRO" = "galactic" ]; then \
+       cd /tmp && curl --silent https://s3.amazonaws.com/RTI/Bundles/5.3.1/Evaluation/rti_connext_dds_secure-5.3.1-eval-x64Linux3gcc5.4.0.tar.gz | tar -xz \
+       && tar -xvf /tmp/openssl-1.0.2n-target-x64Linux3gcc5.4.0.tar.gz; \
```
Comment (nuclearsandwich) [inline linux_docker_resources/Dockerfile-RHEL:126]: It made more sense to me to make this one long multiline RUN than it did to perform the same check on two consecutive RUN lines.

File: linux_docker_resources/Dockerfile (hunk 1 above)
Comment (clalancette) [reply linux_docker_resources/Dockerfile:6]: Agreed, I think we should keep them consistent.  If we also go forth with the plan to make these Dockerfiles install "everything needed to build", then I think we'll need this anyway.

**Comment by nuclearsandwich** [comment]
CI

Rolling [Build Status](https://ci.ros2.org/job/test_ci_linux-rhel/1/)
Galactic [Build Status](https://ci.ros2.org/job/test_ci_linux-rhel/2/)
Foxy [Build Status](https://ci.ros2.org/job/test_ci_linux-rhel/3/)

File: linux_docker_resources/Dockerfile-RHEL (hunk 2 above)
Comment (clalancette) [reply linux_docker_resources/Dockerfile-RHEL:126]: Agreed with that, but isn't this missing a `fi` somewhere?

Comment (nuclearsandwich) [reply linux_docker_resources/Dockerfile-RHEL:126]: sigh. lost in the rebases.

Comment (nuclearsandwich) [reply linux_docker_resources/Dockerfile-RHEL:126]: [9bc507e](https://github.com/ros2/ci/pull/652/commits/9bc507e1ebfdd2b78b79599b880b9f067779d274)

**Comment by nuclearsandwich** [comment]
CI up to test_communication and test_security

Rolling [Build Status](https://ci.ros2.org/job/test_ci_linux-rhel/4/)
Galactic [Build Status](https://ci.ros5.org/job/test_ci_linux-rhel/5/)
Foxy [Build Status](https://ci.ros2.org/job/test_ci_linux-rhel/6/)

**Review by cottsay — APPROVED** [review]
This seems like the right approach.

We should probably align the Ubuntu Dockerfile with this approach. Right now, if we ran a Rolling job on Focal, we'd get the wrong Connext version. Probably save that for a follow-up PR though.

**Comment by nuclearsandwich** [comment]
> We should probably align the Ubuntu Dockerfile with this approach. Right now, if we ran a Rolling job on Focal, we'd get the wrong Connext version. Probably save that for a follow-up PR though.

That's a good point. Done in #653 for Monday.

**Pull request merged**